### PR TITLE
Change Snapshot signature to match simulator.snapshot (#2592)

### PR DESCRIPTION
**Pull request by chowington**
### Summary
This updates the pulse.commands.snapshot.Snapshot class to make its signature align with the signature of extensions.simulator.snapshot.Snapshot (#2592)

### Details and comments
* Param name change: name -> label
* Param name change: snap_type -> snapshot_type
* Make snapshot_type optional with default value of 'statevector'
* New optional param 'name' which defaults to the value of label

### CODE REVIEW

File: qiskit/pulse/commands/snapshot.py [hunk 1]
```diff
@@ -57,8 +69,9 @@ def __eq__(self, other):
             bool: are self and other equal.
         """
         if (type(self) is type(other) and
-                self.name == other.name and
-                self.type == other.type):
+                self.label == other.label and
+                self.type == other.type and
+                self.name == other.name):
```
Comment (taalexander) [inline qiskit/pulse/commands/snapshot.py]: I do not think we need to include `name` in the equality check as this parameter is just used for display purposes.

File: qiskit/qobj/converters/pulse_instruction.py [hunk 2]
```diff
@@ -214,9 +214,9 @@ def convert_snapshot(self, shift, instruction):
             dict: Dictionary of required parameters.
         """
         command_dict = {
-            'name': 'snapshot',
```
Comment (taalexander) [inline qiskit/qobj/converters/pulse_instruction.py]: This is a confusing aspect of the Qobj spec, but the `name` in the terra api is just a reference for display purposes. Within the Qobj transport layer, `name` references the type of command which in this case is `snapshot`. Consequently, this change should be reverted.  See [the spec for more information](https://arxiv.org/abs/1809.03452).

File: qiskit/qobj/converters/pulse_instruction.py [hunk 3]
```diff
@@ -409,4 +409,4 @@ def convert_snapshot(self, instruction):
             Schedule: Converted and scheduled Snapshot
         """
         t0 = instruction.t0
-        return commands.Snapshot(instruction.label, instruction.type) << t0
```
Comment (taalexander) [inline qiskit/qobj/converters/pulse_instruction.py]: This should also be reverted as in this case `name==label`.

**Comment by chowington** [comment]
@taalexander Thanks for the review - the requested changes have been made.

**Comment by chowington** [comment]
The [contribution guidelines](https://github.com/Qiskit/qiskit-terra/blob/master/CONTRIBUTING.md) say that you should add a line to a file named CHANGELOG.rst.  Looks like the file is now CHANGELOG.md - is this something I should do for this PR?

File: qiskit/pulse/commands/snapshot.py [hunk 4]
```diff
@@ -68,5 +82,9 @@ def to_instruction(self):
     # pylint: enable=arguments-differ
 
     def __repr__(self):
-        return '%s(%s, %s) -> %s' % (self.__class__.__name__, self.name,
-                                     self.type, self.channels)
+        if self.label == self.name:
```
Comment (taalexander) [inline qiskit/pulse/commands/snapshot.py]: Maybe we don't need this check and can always print out the name to be more consistent?

**Comment by taalexander** [comment]
Hello @chowington, I believe a new set of tests that were added are now causing an issue with your PR. Could you update it, please and thank you?

File: qiskit/pulse/commands/snapshot.py (hunk 4 above)
Comment (chowington) [reply qiskit/pulse/commands/snapshot.py]: Sure. Changed it to always print the name.

**Comment by chowington** [comment]
@taalexander Done 👍

**Comment by taalexander** [comment]
Thank you very much for your contribution @chowington!

**Pull request merged**